### PR TITLE
Adding IndexCreationStatistics struct to address #95

### PR DESCRIPTION
**Pull request by PaulPyl**
I tried to address #95 but the Github checks fail on me. This is weird since I didn't touch the code in aln.cpp at all.

### CODE REVIEW

**Comment by PaulPyl** [comment]
This should work now.

File: src/index.cpp [hunk 1]
```diff
@@ -323,6 +323,7 @@ IndexCreationStatistics StrobemerIndex::populate(float f) {
     mers_index.reserve(unique_mers);
     // construct index over flat array
     IndexCreationStatistics index_stats = index_vector(h_vector, mers_index, f);
+    filter_cutoff = index_stats.filter_cutoff;
```
Comment (PaulPyl) [inline src/index.cpp:326]: This was the issue ... took me a while to find.

File: src/main.cpp [hunk 2]
```diff
@@ -211,7 +211,33 @@ int run_strobealign(int argc, char **argv) {
     } else {
         // Generate the index
         auto start = high_resolution_clock::now();
-        index.populate(opt.f);
+        IndexCreationStatistics index_creation_stats = index.populate(opt.f);
+        // BEGIN: Logging code moved from index_vector and StrobemerIndex::prune
```
Comment (marcelm) [inline src/main.cpp:215]: Hi Paul, you merged this already, so it’s a bit late, but can you please remove this comment? The information where the code used to be is available in Git history.

File: src/main.cpp [hunk 3]
```diff
@@ -211,7 +211,33 @@ int run_strobealign(int argc, char **argv) {
...
+        }
+        if (index_creation_stats.tot_mid_ab >= 1) {
+            logger.debug() << "Ratio distinct to non distinct: " << index_creation_stats.tot_distinct_strobemer_count / (index_creation_stats.tot_high_ab + index_creation_stats.tot_mid_ab) << std::endl;
+        }
+        logger.debug() << "Filtered cutoff index: " << index_creation_stats.index_cutoff << std::endl;
+        logger.debug() << "Filtered cutoff count: " << index_creation_stats.filter_cutoff << std::endl << std::endl;
+        
+        logger.info() << "Total time generating hash table index: " << index_creation_stats.elapsed_hash_index.count() << " s" <<  std::endl;
+
+        // END: Logging code moved from index_vector and StrobemerIndex::prune
```
Comment (marcelm) [inline src/main.cpp:239]: Same here.

**Comment by marcelm** [comment]
Issue #95 is still open: Remember that you can write "Closes #95" in your commit message to have GitHub auto-close the referenced issue as soon as the commit is merged into main.

**Comment by PaulPyl** [comment]
Right, I just removed these two lines in 0bbadec, I will close the issue manually this time.

**Comment by marcelm** [comment]
Thanks!

**Pull request merged**